atomate/cp2k/fireworks/core.py

```diff
@@ -0,0 +1,191 @@
+
+
+class BaseFW(Firework):
+
+    def __init__(self, structure=None, name=None, cp2k_input_set=None,
+                 cp2k_input_set_params={}, cp2k_cmd="cp2k", prev_calc_loc=True,
+                 prev_calc_dir=None, db_file=None, cp2ktodb_kwargs=None,
+                 parents=None, files_to_copy=[], **kwargs):
+        """
+        Standard calculation Firework. A FW generally follows the following structure:
+            (1) Update based on previous calculation
+            (2) Copy files from previous calculation
+            (3) Write input files as necessary
+            (4) Run Cp2k with custodian
+            (5) Pass this calculation location to the database (in order to stitch calculations together)
+            (6) Run CP2K drone to assimilate results of the calculation
+
+        And this doesn't change, whether its a FW that is part of series of calculations, or a parallel
+        FW, or a standalone. Therefore, this BaseFW defines this general structure, and other FWs simply
+        specify a slightly specific variation on this. Generally, the FWs that inherit will simply specify
+        what the default input set is.
+
+        Args:
+            structure (Structure): Input structure. Note that for prev_calc_loc jobs, the structure
+                is only used to set the name of the FW and any structure with the same composition
+                can be used.
+            name (str): Name for the Firework.
+            cp2k_input_set (Cp2kInputSet): input set to use. Needed unless a previous calculation's inputs
+                parameters will be used.
+            cp2k_input_set_params (dict): Dict of Cp2kInputSet kwargs. Remember, overriding default set parameters
+                is done inside of here using the "{override_default_params: {...} }" as an entry
+            cp2k_cmd (str): Command to run cp2k. Supports env_chk.
+            prev_calc_loc (bool or str): If true (default), copies outputs from previous calc. If
+                a str value, retrieves a previous calculation output by name. If False/None, will create
+                new calculation using the provided structure.
+            prev_calc_dir (str): Path to a previous calculation to copy from.
+            db_file (str): Path to file specifying db credentials. Supports env_chk.
+            parents (Firework): Parents of this particular Firework. FW or list of FWS.
+            cp2ktodb_kwargs (dict): kwargs to pass to Cp2kToDb
+            \*\*kwargs: Other kwargs that are passed to Firework.__init__.
+        """
+        t = []
+
+        # Set up the name, the input set, and how cp2k will be pushed to DB
+        fw_name = "{}-{}".format(structure.composition.reduced_formula if structure else "unknown", name)
+
+        cp2k_input_set = cp2k_input_set or StaticSet(structure, **cp2k_input_set_params)
+        cp2k_input_set.project_name = name  # Currently, this assertion is needed to keep all FWs in a workflow
```

Comment:
What is the problem you are trying to solve by using `project_name`?

Reply:
The CP2K output files are named according to the "PROJECT_NAME" variable inside the input file. Notably, the wavefunction and DOS files have the project name prepended in their filename. So, in order to deal with this we would need to either (a) check for the project name every time you try to find a file (like a restart file) to copy over, (b) use a universal project name so that the files still have the same format but with a standard prefix, (c) override the file name in the cp2k input file so that it adheres to some agreed upon overall file name. I chose to go with (b) as the easiest one, but I'm open to changing it.

Reply:
Note we're going to have the same question for my (perpetual backburner project) of CASTEP support, so interested in whatever you settle on. I might pick the seed name to be the firework id, unless that has issues I'm not anticipating.

Reply:
Ok, thanks for clarifying that. With option b) are you using the same project name for EVERY calculation or is it specific to the calculation type?  Maybe what I am describing is option c?

Are there any downsides to using a fixed project name, and just doing the calculations in different directories? That way there is never any ambiguity over which files to load. it seems like this would remove a lot of the complexity and make the parsing and file copying more similar to how it is handled in the VASP module?

Reply:
Currently option (b) uses the same name for the workflow. So you could name the files according to "Static" for static WFs as an example. Just to give you a sense of what the output files are like. If you did that you would get a directory like this (for Si):

launcher-X:
- Static-ALPHA_k1-1.pdos  # elemental dos for atom kind 1, Spin up channel   
- Static-BETA_k1-1.pdos     # elemental dos for atom kind 1, Spin down channel    
- Static-ALPHA_list1-1.pdos  # Site projected dos onto site list 1, spin up                                   
- Static-ALPHA_list2-1.pdos  # Site projected dos onto site list 2, spin up                                 
- Static-BETA_list1-1.pdos  # Site projected dos onto site list 1, spin down                      
- Static-BETA_list2-1.pdos     # Site projected dos onto site list 2, spin down                                            
- Static-RESTART.wfn     # Restart wavefunction file
- Static-RESTART.wfn.bak-1  # Backup wavefunction file (dumped intermittently)                              
- Static-v_hartree-1_0.cube.  # Hartree (electrostatic potential)
- Static-ELECTRON_DENSITY-1_0.cube   # Electron density      
- Static-SPIN_DENSITY-1_0.cube   # spin density

So you can see that it changes the naming, but its pretty repeatable. Something I've been doing in the pymatgen output parsing is globbing to find files containing, say, "pdos", to identify the pdos files and globbing for "RESTART" to get the wavefunction file name. We could think about doing this instead of trying to record what the file name is going to be.